### PR TITLE
Add missing ethical licenses

### LINKED ISSUE #1898: Ensure all newer "ethical" licenses are added
See https://ethicalsource.dev/licenses/

### PR DESCRIPTION
**Pull request by JRavi2**
Signed-off-by: Ravi <jravi248@gmail.com>

Fixes #1898
I haven't added the Atmosphere Licenses as it is being added in #1998. We're still on hold for Maintainers' Contract license from #1896. The 966.ICU and Hippocratic licenses are already up to date. 
Also, I'm not sure what category these fall into, it would be great if someone could help with that.
Some tests are failing, which could be fixed by replacing “ with " in cal-software.LICENSE and cc-plus-cal.LICENSE but are we allowed to do that?

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

File: src/licensedcode/data/licenses/cal-software.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+key: cal-software
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/cal-software.yml:1]: Can you add the version to this and rename that to this key and file names to `corporate-accountability-1.1`?
There is another unrelated OSI-approved license called cal-1.0.

File: src/licensedcode/data/licenses/cc-plus-cal.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+key: cc-plus-cal
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/cc-plus-cal.yml:1]: Can you add the version to this and rename that to this key and file names to `corporate-accountability-commercial-1.1`?

File: src/licensedcode/data/licenses/cnpl-4.0.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+key: cnpl-4.0
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/cnpl-4.0.yml:1]: Can you rename this to a longer `cooperative-non-violent-4.0` ? the rationale is that less common license are entitled to more verbose license keys so they can stand out.

File: src/licensedcode/data/licenses/do-no-harm.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+key: do-no-harm
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/do-no-harm.yml:1]: Add a version e.g.  `key: do-no-harm-1.0`

File: src/licensedcode/data/licenses/npl-4.0.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+key: npl-4.0
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/npl-4.0.yml:1]: Use instead `non-violent-4.0`

**Comment by JRavi2** [comment]
@pombredanne Thanks for the review, I'll make the suggested changes.

As for the failing tests, [here](https://dev.azure.com/nexB/scancode-toolkit/_build/results?buildId=1832&view=logs&j=f756e756-d784-5c8d-8be7-b20b82a9be96&t=686e8742-372d-5eeb-000f-5b2f296835fe&l=545) is one of them. It's giving UnicodeDecodeError which I'm pretty sure is due to the “ character. Some occurrences of this character can be found in cal-software.LICENSE [here](https://github.com/nexB/scancode-toolkit/pull/2000/files#diff-cd1eb8f5c0b0f514761c339bb259c6e7R7) or in cc-plus-cal.LICENSE [here](https://github.com/nexB/scancode-toolkit/pull/2000/files#diff-87f95475d0d38e5ea7a5e44307558414R3).

File: src/licensedcode/data/licenses/do-no-harm.yml (hunk 4 above)
Comment (JRavi2) [reply src/licensedcode/data/licenses/do-no-harm.yml:1]: A proper version number hasn't been mentioned in this license, only "pre 1.0" is mentioned. They currently are working towards version 1.0. So should I just add `do-no-harm-pre1.0`?

File: src/licensedcode/data/licenses/npl-4.0.yml (hunk 5 above)
Comment (JRavi2) [reply src/licensedcode/data/licenses/npl-4.0.yml:1]: The file should also be renamed right?

**Comment by pombredanne** [comment]
The unicode error is because as you mentioned there are some non-UTF-8-encoded quote `“`. Use plain double quotes instead

**Comment by JRavi2** [comment]
What about the version number for do-not-harm license as asked in the comments? Should I rename the file to `do-not-harm-pre1.0.LICENSE`?

File: src/licensedcode/data/licenses/do-no-harm.yml (hunk 4 above)
Comment (Pratikrocks) [reply src/licensedcode/data/licenses/do-no-harm.yml:1]: @JRavi2  the do-no-harm-license ([link](https://github.com/raisely/NoHarm/blob/publish/licenses/Apache-2.0-NoHarm.md)) is of version 1.0,so I think the name should be `do-no-harm-1.0`

File: src/licensedcode/data/licenses/npl-4.0.yml (hunk 5 above)
Comment (Pratikrocks) [reply src/licensedcode/data/licenses/npl-4.0.yml:1]: Yes the names of the .LICENSE and .yml file are also needed to be changed as per the key name

**Comment by Pratikrocks** [comment]
@JRavi2  in the atmosphere license too , the `" "` character is present ([line](https://github.com/nexB/scancode-toolkit/pull/1998/files#diff-6f030da2a17ff4e562d6aae41c947a5fR44)),but this error is not encountered there.

File: src/licensedcode/data/licenses/do-no-harm.yml (hunk 4 above)
Comment (JRavi2) [reply src/licensedcode/data/licenses/do-no-harm.yml:1]: @Pratikrocks I believe you are referring to a different file. I've added [this](https://github.com/raisely/NoHarm/blob/publish/LICENSE.md) file, as in the readme of the repository, it has been mentioned to use this file.

**Comment by JRavi2** [comment]
@Pratikrocks The double quotes you are referring to are UTF-8 encoded, so you are not recieving any errors.

File: src/licensedcode/data/licenses/npl-4.0.yml (hunk 5 above)
Comment (JRavi2) [reply src/licensedcode/data/licenses/npl-4.0.yml:1]: Okay Thanks :)

File: src/licensedcode/data/licenses/do-no-harm.yml (hunk 4 above)
Comment (Pratikrocks) [reply src/licensedcode/data/licenses/do-no-harm.yml:1]: @JRavi2 .I think it the Readme is not updated, but the current version as per this [link](https://github.com/raisely/NoHarm/blob/publish/licenses/Apache-2.0-NoHarm.md) is 1.0

Comment (JRavi2) [reply src/licensedcode/data/licenses/do-no-harm.yml:1]: The version mentioned there is 0.1. But, I guess it does look like a more updated version of the license. I'll change it here.

**Comment by pombredanne** [comment]
Looking all good and merging at last!
Thanks you ++ for your patience!

**Pull request merged**